configs/cache_partitioning/cache_partitioning.py

```diff
@@ -0,0 +1,218 @@
+# |----------------|---|-----|-----|-----|-----|-----|-----|------|
+# | Cache Hits     | 0 | 256 | 384 | 512 | 640 | 768 | 896 | 1024 |
+#
+# Using the MaxCapacity policy, expected results are the following:
+#
+# | Allocation % | 10 | 20  | 30  | 40  | 50  | 60  | 70  | 80  | 90  | 100  |
+# |--------------|----|-----|-----|-----|-----|-----|-----|-----|-----|------|
+# | Cache Hits   | 0  | 152 | 307 | 409 | 512 | 614 | 716 | 819 | 921 | 1024 |
+
+import argparse
+import os
+import sys
+
+import m5
+from m5.objects import *
+from m5.util import addToPath
+
+# gem5 config imorts
+addToPath("../")
+from common.MemConfig import config_mem
+
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+
+
+STATIC_OPTIONS = argparse.Namespace(
+    mem_channels=1,
+    mem_type="SimpleMemory",
+    # configure values for SimpleMemory interaface
+    request_latency="10ns",
+    mem_bandwidth="1GiB/s",
+    # simulation settings
+    system_frequency="10THz",
+    max_ticks=2000000000,
+)
+
+
+def capacityAllocation(capacity_str):
+    """
+    Verify that Max Capacity partitioning policy has been provided with a suitable
+    configuration
+    """
+    capacity = float(capacity_str)
+
+    if capacity > 1 or capacity < 0:
+        raise argparse.ArgumentTypeError(
+            "Max Capacity Policy needs allocation in range [0, 1]"
+        )
+
+    return capacity
+
+
+def wayAllocation(way_str):
+    """
+    Verify that Way partitioning policy has been provided with a suitable
+    configuration
+    """
+    way_alloc = int(way_str)
+
+    if way_alloc < 0:
+        raise argparse.ArgumentTypeError(
+            "Way Policy needs positive number of ways"
+        )
+
+    return way_alloc
+
+
+def generatePartPolicy(args):
+    """
+    Generate Partitioning Policy object based on provided arguments
+    """
+    assert args.policy in [
+        "way",
+        "max_capacity",
+    ], "Only support generating way and max_capacity policies"
+
+    if args.policy == "way":
+        allocated_ways = [way for way in range(0, args.way_allocation)]
+        allocation = WayPolicyAllocation(partition_id=0, ways=allocated_ways)
+
+        return WayPartitioningPolicy(allocations=[allocation])
+
+    if args.policy == "max_capacity":
+        return MaxCapacityPartitioningPolicy(
+            partition_ids=[0], capacities=[args.capacity_allocation]
+        )
+
+
+def configSystem():
+    """
+    Configure base system and memory
+    """
+
+    system = System(membus=IOXBar(width=128))
+    system.clk_domain = SrcClockDomain(
+        clock=STATIC_OPTIONS.system_frequency,
+        voltage_domain=VoltageDomain(),
+    )
+
+    # add memory
+    mem_range = AddrRange("128MB")
+    system.mem_ranges = [mem_range]
+
+    # Memory configuration
+    config_mem(STATIC_OPTIONS, system)
+    system.mem_ctrls[0].latency = STATIC_OPTIONS.request_latency
+    system.mem_ctrls[0].bandwidth = STATIC_OPTIONS.mem_bandwidth
```

Comment:
Just hardcode this with the `SimpleMemory`. There is no need for extra indirections into the `config_mem` function.